test/query_component_test.dart

```diff
@@ -72,17 +77,137 @@ void main() {
 
   tearDown(disposeAnyRunningTest);
 
-  group('populate button clicks invokes QueryService function:', () {
-    setUp(() async => await queryComponentPO.clickPopulate());
-
-    test('execDV3Query', () => verify(mockQueryService.execDV3Query(any, any)));
-
-    test('execReportingCreateQuery',
-        () => verify(mockQueryService.execReportingCreateQuery(any, any, any)));
+  group('radio button clicks change QueryType:', () {
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByAdvertiser());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byAdvertiser, '', null, null, null));
+    });
+
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByMediaPlan());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byMediaPlan, '', null, null, null));
+    });
+
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByIO());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byInsertionOrder, '', null, null, null));
+    });
+
+    test('only the last selection matters', () async {
+      await (queryComponentPO.selectByIO());
+      await (queryComponentPO.selectByAdvertiser());
+      await (queryComponentPO.selectByMediaPlan());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
```

Comment:
what if i hit populate without making a radio button selection? is there a default value?

Reply:
The default value is now set to byAdvertiser.

Reply:
should add test for that